components/brave_rewards/resources/ui/components/adsBox.tsx

```diff
@@ -28,14 +36,66 @@ class AdsBox extends React.Component {
     )
   }
 
+  onAdsSettingChange = (key: string, value: string) => {
+    let newValue: any = value
+    const { adsEnabled } = this.props.rewardsData.adsData
+
+    if (key === 'adsEnabled') {
+      newValue = !adsEnabled
+    }
+
+    this.props.actions.onAdsSettingSave(key, newValue)
+  }
+
+  adsSettings = (enabled?: boolean) => {
+    if (!enabled) {
+      return null
+    }
+
+    const { adsPerHour } = this.props.rewardsData.adsData
+
+    return (
+      <Grid columns={1} customStyle={{ maxWidth: '270px', margin: '0 auto' }}>
+        <Column size={1} customStyle={{ justifyContent: 'center', flexWrap: 'wrap' }}>
+          <ControlWrapper text={getLocale('adsPerHour')}>
+            <Select
+              value={adsPerHour.toString()}
+              onChange={this.onAdsSettingChange.bind(this, 'adsPerHour')}
+            >
+              {['1', '2', '3', '4', '5'].map((num: string) => {
+                return (
+                  <div key={`num-per-hour-${num}`} data-value={num}>
+                   {getLocale(`adsPerHour${num}`)}
+                  </div>
+                )
+              })}
+            </Select>
+          </ControlWrapper>
+        </Column>
+      </Grid>
+    )
+  }
+
   render () {
+    const { adsData } = this.props.rewardsData
+
+    if (!adsData) {
+      return null
```

Comment:
I think we should displayed disabled state (like we currently have in the production) with text if we don't have adsData and not just hide the box